r/inst/build_arrow_static.sh

```diff
@@ -54,7 +67,7 @@ ${CMAKE} -DARROW_BOOST_USE_SHARED=OFF \
     -DARROW_COMPUTE=ON \
     -DARROW_CSV=ON \
     -DARROW_DATASET=${ARROW_DATASET:-ON} \
-    -DARROW_DEPENDENCY_SOURCE=BUNDLED \
+    -DARROW_DEPENDENCY_SOURCE=${ARROW_DEPENDENCY_SOURCE} \
```

Comment:
Hmmm, something is going on and this isn't getting passed through, it looks like `DEPENDENCY_SOURCE` is still set to bundled:

https://github.com/ursacomputing/crossbow/runs/3170999772?check_suite_focus=true#step:8:283

I'm still tracking down where this is falling out though, so don't yet have an explanation

Reply:
That's weird; I did a diff comparing the `BUNDLED` and `AUTO` jobs, and the main thing that stood out to me there was that in the `AUTO` one, there's a load of lines that start with "Looking for" and then whatever dependency, so I assumed it was working.  

Could this be related to @nealrichardson 's comment below about it needing to be done earlier, in `r/configure`?

Reply:
Yeah, partially: I suspect that there were builds where it was triggered because the names of the variables lined up just right and that's where you saw that, but at least on the latest build I'm seeing the same number of "Looking for" on both.

But also like neal mentioned up top, setting the env var with `-e` for the runtime only will make sure it gets in the place we need it without having to worry about passing it around. (I'm going to add a bit more to my comment up there since it's mostly related to that...)

Reply:
OK, deleted these changes now then

Reply:
Let's leave a default value here so that the script doesn't depend on being called from nixlibs.R

```suggestion
    -DARROW_DEPENDENCY_SOURCE=${ARROW_DEPENDENCY_SOURCE:-BUNDLED} \
```